File: 백준/Silver/1543. 문서 검색/문서 검색.cc
#include <iostream>
using namespace std;
int main(void) {
    int i = 0, cnt = 0;
    string st, tar, t;
    getline(cin, st);
    getline(cin, tar);
    if (tar.size() <= st.size()) {
        while (i < st.size() - tar.size() + 1) {
            t = st.substr(i, tar.size());
            if (t == tar) {
                i += tar.size();
                cnt += 1;
            } else {
                i += 1;
            }
        }
    } else {
        cnt = 0;
    }
    cout << cnt << "\n";
    return 0;
}

File: 백준/Silver/1543. 문서 검색/문서 검색.py
import sys
input=lambda:sys.stdin.readline().rstrip()
st=input()
tar=input()
i,cnt=0,0
while i<=len(st)-len(tar):
    if st[i:i+len(tar)]==tar:
        cnt+=1
        i+=len(tar)
    else:
        i+=1
print(cnt)